packages/rum-core/test/performance-monitoring/transaction-service.spec.js

```diff
@@ -758,15 +762,36 @@ describe('TransactionService', function() {
         managed: true,
         canReuse: true
       })
-      expect(startSpy).toHaveBeenCalledWith('longtask')
+      expect(startSpy).toHaveBeenCalledWith(LONG_TASK)
       const managedNonReusable = trService.startTransaction('test', 'custom', {
         managed: true
       })
-      expect(startSpy.calls.allArgs()).toEqual([['longtask'], ['longtask']])
+      expect(startSpy.calls.allArgs()).toEqual([[LONG_TASK], [LONG_TASK]])
       expect(stopSpy).toHaveBeenCalled()
       await managedReusable.end()
       await managedNonReusable.end()
       expect(stopSpy).toHaveBeenCalledTimes(2)
     })
+
+    it('should not record longtasks when monitorLongtasks is false', async () => {
+      config.setConfig({
+        monitorLongtasks: false
+      })
+      const pageLoadTr = trService.startTransaction('test', PAGE_LOAD, {
+        managed: true
+      })
+      expect(startSpy).toHaveBeenCalledTimes(1)
```

Comment:
I think it would be better to recreate the spy in `beforeEach` for this `describe` set instead of reseting it every time. 

Also, in the `beforeEach` we can set the `recorder` object as well

Reply:
We need to reset spies in the middle of tests as well https://github.com/elastic/apm-agent-rum-js/pull/600/files#diff-984832cc3eb80072b4dcfa9401a47af1L728 not just always in the end or the start, Would this change help us? 

I can move the resetSpeis to beforeEach, I just feel both would do the same job.